transport/response.go

```diff
@@ -24,5 +24,6 @@ import "io"
 
 // Response is the low level response representation.
 type Response struct {
-	Body io.ReadCloser
+	Headers map[string]string
+	Body    io.ReadCloser
```

Comment:
True. I was planning on adding a fixed `transport.Properties` struct to requests and responses similar to tchannel-python (and tchannel-go). It will contain a fixed set of fields which represent the different transport headers we support. It will be up to the transport to translate them into transport-level headers.

I'm open to naming suggestions besides transport properties. CC @breerly